zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/ProcessInstanceStateTransitionGuard.java

```diff
@@ -207,6 +209,22 @@ private Either<String, ElementInstance> hasNonInterruptedFlowScope(
     }
   }
 
+  private Either<String, ?> hasActiveProcessInstance(final BpmnElementContext context) {
+    if (context.getBpmnElementType() == BpmnElementType.PROCESS) {
+      return Either.right(null);
+    }
+    final var processInstance = stateBehavior.getElementInstance(context.getProcessInstanceKey());
+    if (processInstance.isInterrupted()
+        && !processInstance.getInterruptingElementId().equals(context.getElementId())) {
```

Comment:
❓ What purpose does the right hand of the `&&` in the if statement have?

Reply:
Good question, it's actually not needed because the interrupting element is never supposed to complete in our case 👍